core/src/ops/matmul/mir_quant.rs

```diff
@@ -156,14 +156,17 @@ impl EvalOp for QMatMul {
         let b = model.add_const("source_b", inputs[1].clone())?;
         let bias = model.add_const("source_bias", inputs[2].clone())?;
 
-        let params = self
+        let mut params = self
             .params
             .iter()
             .map(|(name, qp)| {
                 model.add_const(format!("source_{}", name), qp.tensor(&inputs).clone())
             })
             .collect::<TractResult<Vec<_>>>()?;
 
+        let a = wire_offset_u8_as_i8(&mut model, "adhoc", a, "a", &mut params[0], "a0")?;
+        let b = wire_offset_u8_as_i8(&mut model, "adhoc", b, "b", &mut params[2], "b0")?;
```

Comment:
This might be completely wrong (as I don't understand too much what is going on) but:

```suggestion
        let b = wire_offset_u8_as_i8(&mut model, "adhoc", b, "b", &mut params[2], "b0")?;
        let bias = wire_offset_u8_as_i8(&mut model, "adhoc", bias, "bias", &mut params[4], "bias0")?;
```

seems like it could make sense.

Reply:
how to handle the `bias` term is a bit unclear to me as well. it is always of type `i32` and the way it is used in `wire_matmul_quant()` means that it is only affected by the scales but not the zero points (i would assume that the `bias` is expected to be given in a scaled form, but this would affect the computation of the scales which i didn't observe, so maybe it is compensated via the `c_scale`). conversely, we probably also don't have to apply `wire_offset_u8_as_i8()` to it, which would also have no effect on it in its current implementation.

unfortunatetly, i didn't find any examples in the tract code except for `None`/`0` bias. we definitely should take this question to the tract guys later on.